src/nifreeze/model/dmri.py

```diff
@@ -173,16 +174,24 @@ class AverageDWIModel(ExpectationModel):
 
     __slots__ = ("_th_low", "_th_high", "_detrend")
 
-    def __init__(self, dataset, stat="median", th_low=100, th_high=100, detrend=False, **kwargs):
+    def __init__(
+        self,
+        dataset: DWI,
+        stat: str = "median",
+        th_low: int = 100,
+        th_high: int = 100,
+        detrend: bool = False,
+        **kwargs,
+    ):
         r"""
         Implement object initialization.
 
         Parameters
         ----------
-        th_low : :obj:`numbers.Number`
+        th_low : :obj:`int`
```

Comment:
I don't think we want to limit this to only integers, float would be just fine.

Reply:
Not sure I've ever seen a non-integer b-value, but will change it to `float`.

We may want to reduce the precision here:
https://github.com/nipreps/nifreeze/pull/72/files#diff-774e30bff24e702cda5516cbc6bbbcae1af21b710e2b0ad22cc15741fbed9afeR250

6 decimal places seems a lot, and makes reading the b-val file difficult for a human reader. Maybe fmt="%.1f" or fmt="%.2f" at most?

Reply:
This is not expressing b-values, but excursions from a central b-value. I agree there's probably no difference between integer and float here, but writing float ensures no hidden rounding errors happen.

Re: decimal places. Yes, that doesn't look great. Perhaps we should add an argument for the decimal places and set it to zero for the default.

Reply:
> Re: decimal places. Yes, that doesn't look great. Perhaps we should add an argument for the decimal places and set it to zero for the default.

Commit 138573f. Resolving.